src/js/actions/menu.js

```diff
@@ -63,18 +74,38 @@ define(function (require, exports) {
             return Promise.reject(error);
         }
 
-        // Photoshop expects commandId with a lower case d, so convert here
-        payload.commandId = payload.commandID;
-
-        return ps.performMenuCommand(payload).then(function (success) {
-            // FIXME: After the M2 release, remove the debug conjunct
-            if (global.debug && !success) {
-                throw new Error("Menu command not available: " + payload.commandID);
-            }
-        });
+        var isPlaceCommand = payload.commandID === _PLACE_LINKED_MENU_ID ||
+                payload.commandID === _PLACE_EMBEDDED_MENU_ID;
+        
+        return Promise.bind(this)
+            .then(function () {
+                if (isPlaceCommand) {
+                    this.dispatchAsync(events.menus.PLACE_COMMAND, { executing: true });
+                    
+                    return this.transfer(policyActions.suspendAllPolicies);
+                }
+            })
+            .then(function () {
+                // Photoshop expects commandId with a lower case d, so convert here
+                payload.commandId = payload.commandID;
+                
+                return ps.performMenuCommand(payload);
+            })
+            .then(function (success) {
+                // FIXME: After the M2 release, remove the debug conjunct
+                if (global.debug && !success) {
+                    throw new Error("Menu command not available: " + payload.commandID);
+                }
+            })
+            .catch(function () {
+                if (!isPlaceCommand) {
```

Comment:
Yes, but this says `if (!placeCommand)`. I understand that we want to clean up after suspending policies as a result of  place command. But it seems like this will do the cleanup only after executing something _other than_ a place command. Sorry if I'm just not getting it :)